udf/crc32_uuid.sql

```diff
@@ -0,0 +1,108 @@
+/*
+Calculate the CRC-32 hash of a 36-byte UUID.
+
+This implementation is limited to an exact length because
+recursion does not work.
+
+Based on https://stackoverflow.com/a/18639999/1260237
+See https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+*/
+CREATE TEMP FUNCTION udf_crc32_table() AS (
+  -- precalculated from this algorithm:
+  -- for c in range(256):
+  --   for _ in range(8):
+  --     c = (0xEDB88320 if c & 1 else 0) ^ (c >> 1)
+  --   yield c
+  [
+    0, 1996959894, 3993919788, 2567524794, 124634137, 1886057615, 3915621685,
```

Comment:
this isn't formatted compatibly with the sql style guide because that makes it unreasonably long